File: src/me/sullivan/glasslang/parser/nodes/NumberNode.java
package me.sullivan.glasslang.parser.nodes;

import me.sullivan.glasslang.interpreter.Interpreter;
import me.sullivan.glasslang.interpreter.primitives.Primitive;
import me.sullivan.glasslang.lexer.token.Token;

public class NumberNode extends Node {

	public NumberNode(Token token)
	{
		super(token);
	}

	@Override
	public Double getValue()
	{
		return Double.parseDouble(token.getValue());
	}

	@Override
	public String toString()
	{
		return "Number(" + getValue() + ")";
	}

	@Override
	public Primitive<?> visitor(Interpreter interpreter)
	{
		return interpreter.visitNumberNode(this);
	}
}

File: src/me/sullivan/glasslang/parser/nodes/NumberNode.kt
package me.sullivan.glasslang.parser.nodes

import me.sullivan.glasslang.interpreter.Interpreter
import me.sullivan.glasslang.interpreter.primitives.Primitive
import me.sullivan.glasslang.lexer.token.Token

class NumberNode(token: Token) : Node(token)
{
    override val value: Double
        get() = token.value.toDouble()

    override fun toString(): String
    {
        return "Number($value)"
    }

    override fun visitor(interpreter: Interpreter): Primitive<*>
    {
        return interpreter.visitNumberNode(this)
    }
}
